### PR TITLE
make-dist: add liburing to dist tarball

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by badone — APPROVED** [review]
Thanks Kefu, Looks good to me.

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-make-dist-kefu/68eb5bcdce8d8e2000ec84a53046bd68f2261860/

**Comment by smithfarm** [comment]
This appears to have broken master `make-dist` on SUSE:

```
-> "boost_1_73_0.tar.bz2" [1]
2020-11-23 15:21:32 URL:https://codeload.github.com/axboe/liburing/tar.gz/liburing-0.7 [111033] -> "liburing-0.7.tar.gz" [1]
tar (child): gzip: Cannot exec: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
```

**Comment by smithfarm** [comment]
I guess the failure shown in the previous comment is a transient one that falls under the category "github noise". But it's hard to tell when `make-dist` does not run with `-x`. I opened https://github.com/ceph/ceph/pull/38244 which, if/when merged, will provide more information the next time my master-RPM-building cronjob hits this.

File: make-dist [hunk 1]
```diff
@@ -68,6 +76,25 @@ download_boost() {
...
 }
 
+download_liburing() {
+    liburing_version=$1
+    shift
+    liburing_sha256=$1
+    shift
+    liburing_fname=liburing-${liburing_version}.tar.gz
+    download_from $liburing_fname $liburing_sha256 $*
+    tar xzf $liburing_fname -C src  \
```
Comment (smithfarm) [inline make-dist:86]: On SUSE, I think this `tar xzf` does require the package `gzip` to be installed in the build environment.

**Comment by smithfarm** [comment]
So, the issue I had with this was/is:

1. I run `make-dist` in a minimal container which used to have everything that was needed to run `make-dist`
2. As of when this PR was merged, introducing a new "tarball generation dependency" (gzip), the container no longer has everything needed.

After adding `gzip` to the list of packages to be installed in the container and regenerating the container image, I believe it should work again.

**Comment by badone** [comment]
@smithfarm So IIRC the problem this PR was trying to solve was an introduced dependency on 'git' so we appear to be caught between a rock and a hard place? Your SUSE container has bzip2 but not gzip? On all the distro containers I checked gzip seems to be installed.

**Comment by badone** [comment]
> I guess the failure shown in the previous comment is a transient one that falls under the category "github noise". But it's hard to tell when `make-dist` does not run with `-x`. I opened #38244 which, if/when merged, will provide more information the next time my master-RPM-building cronjob hits this.

Couldn't you just change your cron job to run the script with 'bash -x'?

**Comment by smithfarm** [comment]
> Couldn't you just change your cron job to run the script with 'bash -x'?

Yes. That will help me, but no one else.

**Pull request merged**